File: Reactive/example/src/main/java/com/didan/reactive/learn/r2dbc/entity/Customer.java
package com.didan.reactive.learn.r2dbc.entity;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.relational.core.mapping.Column;
import org.springframework.data.relational.core.mapping.Table;

@Table("customer") // Map class Customer đên bảng "customer" trong cơ sở dữ liệu (Trong R2DBC chỉ có @Table, không có @Entity như JPA)
@NoArgsConstructor
@AllArgsConstructor
@Data
public class Customer {

  @Id
  private Integer id;

  @Column("name")
  private String name;

  @Column("email")
  private String email;
}

File: GraphQL/src/main/kotlin/com/didan/demo/graphql/validate_input_error_handling/entity/Customer.kt
package com.didan.demo.graphql.validate_input_error_handling.entity

import org.springframework.data.annotation.Id

data class Customer(
    @Id
    var id: Int? = null,
    var name: String? = null,
    var age: Int? = null,
    var city: String? = null
)